src/main/java/seedu/bookmark/commons/util/StringUtil.java

```diff
@@ -74,7 +75,7 @@ public static boolean isNonZeroUnsignedInteger(String s) {
      */
     public static String trimLeadingZeroes(String s) {
         requireNonNull(s);
-        int value = Integer.parseInt(s);
+        BigInteger value = new BigInteger(s);
```

Comment:
I'm not sure if this needs to be here as you mentioned in the PR that you were using the regex solution instead